leetcodePython/BinarySearch/bs_354.py

```diff
@@ -0,0 +1,26 @@
+import bisect
+from typing import List
+
+
+class Solution:
+    @staticmethod
+    def _lengthOfLIS(nums: List[int]) -> int:
+        sequence = []
+        for num in nums:
+            idx = bisect.bisect_left(sequence, num)
+            if idx == len(sequence):
+                sequence.append(num)
+            else:
+                sequence[idx] = num
+        return len(sequence)
+
+    def maxEnvelopes(self, envelopes: List[List[int]]) -> int:
+        # when width is same, we make height in reverse order, so width same data can not form increasing sequence for height at all.
+        envelopes.sort(key=lambda x: (x[0], -x[1]))
```

Comment:
when width is same, we make height in reverse order, so width same data can not form increasing sequence for height at all.